File: biscuits/src/main/biscuits-ui/src/app/missions/model/Mission.ts
import { Kid } from "src/app/kids/model/Kid";

export class Mission {
  private _id: number;
  private _action: string;
  private _imageURL: string;
  private _isDone: boolean;
  private _biscuitsToEarn: number;
  private _kid : Kid;

  constructor(action: string, biscuitsToEarn: number, kidNickname: string) {
    this._action = action;
    this._biscuitsToEarn = biscuitsToEarn;
    this._kid = new Kid(kidNickname, null);
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get action(): string {
    return this._action;
  }

  set action(value: string) {
    this._action = value;
  }

  get imageURL(): string {
    return this._imageURL;
  }

  set imageURL(value: string) {
    this._imageURL = value;
  }

  get isDone(): boolean {
    return this._isDone;
  }

  set isDone(value: boolean) {
    this._isDone = value;
  }

  get biscuitsToEarn(): number {
    return this._biscuitsToEarn;
  }

  set biscuitsToEarn(value: number) {
    this._biscuitsToEarn = value;
  }

  get kid(): Kid {
    return this._kid
  }

  set kid(value: Kid) {
    this._kid = value;
  }
}

File: biscuits/src/main/java/com/agathevaisse/biscuits/domain/mission/Mission.java
package com.agathevaisse.biscuits.domain.mission;

import com.agathevaisse.biscuits.domain.kid.Kid;
import java.util.Objects;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

public class Mission {

    private Long id;

    @NotBlank
    @Size(min = 3, max = 50)
    private String action;
    private String imageURL;
    private boolean isDone;

    @NotBlank
    @Min(1)
    private int biscuitsToEarn;
    
    @NotBlank
    private Kid kid;

    public Mission() {
    }

    public Mission(String action, int biscuitsToEarn, Kid kid) {
        this.action = action;
        this.imageURL = "src/main/biscuits-ui/src/assets/images/secret-mission.jpg";
        this.isDone = false;
        this.biscuitsToEarn = biscuitsToEarn;
        this.kid = kid;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }

    public String getImageURL() {
        return imageURL;
    }

    public void setImageURL() {
        this.imageURL = "src/main/biscuits-ui/src/assets/images/secret-mission.jpg";
    }

    public boolean isDone() {
        return isDone;
    }

    public void setDone(boolean done) {
        isDone = done;
    }

    public int getBiscuitsToEarn() {
        return biscuitsToEarn;
    }

    public void setBiscuitsToEarn(int biscuitsToEarn) {
        this.biscuitsToEarn = biscuitsToEarn;
    }

    public Kid getKid() {
        return kid;
    }

    public void setKid(Kid kid) {
        this.kid = kid;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Mission)) return false;
        Mission mission = (Mission) o;
        return isDone() == mission.isDone() &&
                getBiscuitsToEarn() == mission.getBiscuitsToEarn() &&
                Objects.equals(getId(), mission.getId()) &&
                Objects.equals(getAction(), mission.getAction()) &&
                Objects.equals(getImageURL(), mission.getImageURL()) &&
                Objects.equals(getKid(), mission.getKid());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), getAction(), getImageURL(), isDone(), getBiscuitsToEarn(), getKid());
    }

    @Override
    public String toString() {
        return "Mission{" +
                "id=" + id +
                ", action='" + action + '\'' +
                ", imageURL='" + imageURL + '\'' +
                ", isDone=" + isDone +
                ", biscuitsToEarn=" + biscuitsToEarn +
                ", kid=" + kid +
                '}';
    }
}
